okio/src/main/java/okio/Buffer.java

```diff
@@ -1253,6 +1253,28 @@ Segment writableSegment(int minimumCapacity) {
     return -1L;
   }
 
+  @Override public long indexOf(ByteString bytes) throws IOException {
+    return indexOf(bytes, 0);
+  }
+
+  @Override public long indexOf(ByteString bytes, long fromIndex) throws IOException {
+    int bytesSize = bytes.size();
+    loop:
+    while ((fromIndex = indexOf(bytes.getByte(0), fromIndex)) != -1) {
```

Comment:
If you think there's too much going on in this line I can switch to a `while (true)` loop and move it to an `if` check. That was in my original implementation.